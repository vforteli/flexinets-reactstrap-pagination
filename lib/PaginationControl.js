module.exports=function(e){var t={};function n(r){if(t[r])return t[r].exports;var a=t[r]={i:r,l:!1,exports:{}};return e[r].call(a.exports,a,a.exports,n),a.l=!0,a.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var a in e)n.d(r,a,function(t){return e[t]}.bind(null,a));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=0)}([function(e,t,n){"use strict";Object.defineProperty(t,"__esModule",{value:!0});var r=function(){function e(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(t,n,r){return n&&e(t.prototype,n),r&&e(t,r),t}}(),a=i(n(1)),o=i(n(2)),u=n(3);function i(e){return e&&e.__esModule?e:{default:e}}function c(e,t){if(!e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!t||"object"!=typeof t&&"function"!=typeof t?e:t}var p=function(e){function t(){var e,n,r;!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t);for(var a=arguments.length,o=Array(a),u=0;u<a;u++)o[u]=arguments[u];return n=r=c(this,(e=t.__proto__||Object.getPrototypeOf(t)).call.apply(e,[this].concat(o))),r.pageChanged=function(e){console.debug("Page changed to: "+e),r.props.pageChanged&&r.props.pageChanged(e)},r.getPageCount=function(){return Math.ceil(r.props.totalCount/r.props.pageSize)},r.getPagesToShow=function(){if(r.getPageCount()<=r.props.maxSize)return Array.from({length:r.getPageCount()},function(e,t){return t+1});var e=Math.ceil(r.props.maxSize/2)-1,t=Math.max(0,1-(r.props.currentPage-e))+Math.min(0,r.getPageCount()-(r.props.currentPage+e+1-r.props.maxSize%2)),n=r.props.currentPage-e;return Array.from({length:r.props.maxSize},function(e,r){return r+n+t})},c(r,n)}return function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function, not "+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}(t,a.default.Component),r(t,[{key:"render",value:function(){var e=this;return a.default.createElement(u.Pagination,null,this.props.maxSize<this.getPageCount()&&a.default.createElement(u.PaginationItem,{disabled:1===this.props.currentPage},a.default.createElement(u.PaginationLink,{href:"",onClick:function(){return e.pageChanged(1)}},"First")),a.default.createElement(u.PaginationItem,{disabled:1===this.props.currentPage},a.default.createElement(u.PaginationLink,{href:"",onClick:function(){return e.pageChanged(e.props.currentPage-1)}},"Previous")),this.getPagesToShow().map(function(t){return a.default.createElement(u.PaginationItem,{key:t,active:e.props.currentPage===t},a.default.createElement(u.PaginationLink,{href:"",onClick:function(){return e.pageChanged(t)}},t))}),a.default.createElement(u.PaginationItem,{disabled:this.props.currentPage===this.getPageCount()},a.default.createElement(u.PaginationLink,{href:"",onClick:function(){return e.pageChanged(e.props.currentPage+1)}},"Next")),this.props.maxSize<this.getPageCount()&&a.default.createElement(u.PaginationItem,{disabled:this.props.currentPage===this.getPageCount()},a.default.createElement(u.PaginationLink,{href:"",onClick:function(){return e.pageChanged(e.getPageCount())}},"Last")))}}]),t}();p.defaultProps={maxSize:10},p.propTypes={totalCount:o.default.number.isRequired,pageSize:o.default.number.isRequired,currentPage:o.default.number.isRequired,pageChanged:o.default.func,maxSize:o.default.number},t.default=p},function(e,t){e.exports=require("react")},function(e,t){e.exports=require("prop-types")},function(e,t){e.exports=require("reactstrap")}]);