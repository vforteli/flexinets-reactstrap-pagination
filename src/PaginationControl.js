import React from 'react';
import PropTypes from 'prop-types';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';


class PaginationControl extends React.Component {

    pageChanged = (page) => {
        console.debug(`Page changed to: ${page}`);
        if (this.props.pageChanged) {
            this.props.pageChanged(page);
        }
    }


    getPageCount = () => Math.ceil(this.props.totalCount / this.props.pageSize);

    getPagesToShow = () => {
        if (this.getPageCount() <= this.props.maxSize) {
            return Array.from({ length: this.getPageCount() }, (v, k) => k + 1);
        }

        // not my proudest moment
        let foo = Math.floor(this.props.maxSize / 2);
        let derp = 0;
        if (this.props.maxSize % 2 === 0) {
            foo--;
            derp = 1;
        }
        const offset = Math.max(0, 1 - (this.props.currentPage - foo)) + Math.min(0, this.getPageCount() - (this.props.currentPage + foo + derp));
        const start = this.props.currentPage - foo;

        return Array.from({ length: this.props.maxSize }, (v, k) => k + start + offset);
    }


    render() {
        return (
            <Pagination>
                {this.props.maxSize < this.getPageCount() &&
                    <PaginationItem disabled={this.props.currentPage === 1}>
                        <PaginationLink href="" onClick={() => this.pageChanged(1)}>First</PaginationLink>
                    </PaginationItem>
                }
                <PaginationItem disabled={this.props.currentPage === 1}>
                    <PaginationLink href="" onClick={() => this.pageChanged(this.props.currentPage - 1)}>Previous</PaginationLink>
                </PaginationItem>

                {this.getPagesToShow().map(i =>
                    <PaginationItem key={i} active={this.props.currentPage === i}>
                        <PaginationLink href="" onClick={() => this.pageChanged(i)}>{i}</PaginationLink>
                    </PaginationItem>
                )}

                <PaginationItem disabled={this.props.currentPage === this.getPageCount()}>
                    <PaginationLink href="" onClick={() => this.pageChanged(this.props.currentPage + 1)}>Next</PaginationLink>
                </PaginationItem>
                {this.props.maxSize < this.getPageCount() &&
                    <PaginationItem disabled={this.props.currentPage === this.getPageCount()}>
                        <PaginationLink href="" onClick={() => this.pageChanged(this.getPageCount())}>Last</PaginationLink>
                    </PaginationItem>
                }
            </Pagination>

        )
    }
}

PaginationControl.defaultProps = {
    maxSize: 9
};

PaginationControl.propTypes = {
    totalCount: PropTypes.number.isRequired,
    pageSize: PropTypes.number.isRequired,
    currentPage: PropTypes.number.isRequired,
    pageChanged: PropTypes.func,
    maxSize: PropTypes.number
};

export default PaginationControl;