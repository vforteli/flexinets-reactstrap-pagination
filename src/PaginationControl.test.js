import React from 'react';
import ReactDOM from 'react-dom';
import PaginationControl from './PaginationControl';

it('shows pagination control without exploding', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PaginationControl loading={true} />, div);
    ReactDOM.unmountComponentAtNode(div);
});
