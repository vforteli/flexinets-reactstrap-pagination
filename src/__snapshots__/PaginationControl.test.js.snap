// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationControl renders correctly with even max size 1`] = `
<div>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`PaginationControl renders correctly with odd max size 1`] = `
<div>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
  <nav
    aria-label="pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          First
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Previous
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item active"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          10
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Next
        </button>
      </li>
      <li
        className="page-item disabled"
      >
        <button
          aria-label={undefined}
          className="page-link"
          href=""
          onClick={[Function]}
        >
          Last
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
